### PR TITLE
cmd/swarm: Add --httpaddr flag

### LINKED ISSUE #14474: Swarm HTTP API is exposed; should be localhost only.
#### Expected behaviour

By default, the swarm HTTP API should be exposed to localhost only. Default: `http://localhost:8500`

#### Actual behaviour

Actually it seems as if the swarm daemon answers any requests coming in on port 8500 such as `http://192.168.1.2:8500`

Suggested actions:

1. Immediately change swarm to listen on localhost only
2. In future possibly add support for a `--listen` cli flag allowing users to specify which IPs to listen on.

### PR DESCRIPTION
**Pull request by lmars**
Fixes #14474.

### CODE REVIEW

File: swarm/api/config.go [hunk 1]
```diff
@@ -48,12 +48,13 @@ type Config struct {
...
 	Swap *swap.SwapParams
 	*network.SyncParams
-	Path      string
-	Port      string
-	PublicKey string
-	BzzKey    string
-	EnsRoot   common.Address
-	NetworkId uint64
+	Path       string
+	ListenAddr string
```
Comment (zelig) [inline swarm/api/config.go]: so you need to add this to the config tests with the default value, it fails the CI test https://travis-ci.org/ethereum/go-ethereum/jobs/232519067#L1592

Comment (lmars) [inline swarm/api/config.go]: Fixed in c1da40213800575aee9d868f93e2dddedf739891.

**Comment by homotopycolimit** [comment]
Maybe the flag should have a different name. The word 'address' is a little overloaded and bzzaddr sounds like it could be the swarm address of the node.

In many servers and daemons the corresponding value is called `--listen` or `--listenaddress`;  maybe we could use that. Or otherwise maybe `--http-address` ?

just a thought.

**Comment by lmars** [comment]
@zelig @homotopycolimit I have renamed to `--httpaddr`.

**Pull request merged**